src/frontend/utils/widgets/widgets.ts

```diff
@@ -0,0 +1,14 @@
+import { Video } from 'types/tracks';
+import { Maybe } from 'utils/types';
+
+let timeoutId: Maybe<number>;
+
+export const debounceVideo = (
```

Comment:
```suggestion
export const debounce = <T>(
```

Now your `debounce` is generic and can be used with any resources like this:

```
debounce<Video>((updatedVideoProperty) => {
      if (updatedVideoProperty.title === '') {
        toast.error(intl.formatMessage(messages.updateTitleBlank), {
          position: 'bottom-center',
        });
        setTitle(video.title);
        return;
      }
      videoMutation.mutate(updatedVideoProperty);
    });
```

Reply:
ty, that's a more elegant way to do it. Done.